### PR TITLE
Implement Fizz Buzz

### PR DESCRIPTION
**Pull request by wtitterton**

### CODE REVIEW

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { fizzBuzz } from "./fizzbuzz";
 
 describe("fizzbuzz", () => {
 
-    test.each([3,9,42])("should return 'Fizz' when given number devisable by 3", (number) => {
+    test.each([3, 9, 42])("should return 'Fizz' when given number devisable by 3", (number) => {
```
Comment (taha-crafter) [inline 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts:5]: I think it's nice and makes your test results more readable if you add a placeholder for the example passed in using %i, like this:

```typescript
test.each([3, 9, 42])(
  "should return 'Fizz' when given number devisable by 3, such as %i",
  number => {
    expect(fizzBuzz(number)).toBe('Fizz');
  }
);`

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts [hunk 2]
```diff
@@ -1,37 +1,25 @@
...
-        
-        expect(() => {fizzBuzz(-12)}).toThrow(message);
-        expect(() => {fizzBuzz(102)}).toThrow(message);
+    test.each([43, 8, 58])("should return a number as a string", (number) => {
+        expect(typeof fizzBuzz(number)).toBe("string");
     });
 
-    it("should throw and error when a non number is provided", () => {
-         const message = "Please provide a number";
-         expect(() => {fizzBuzz("non number")}).toThrow(message);
```
Comment (taha-crafter) [inline 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts]: I'm wondering why you replaced this test with the test of 1-100? isn't it better to make the function explicitly tell when the input type is not accepted? something like this:

```typescript
function fizzbuzz(n: number): string {
  if (typeof n !== "number") {
    throw new Error("Input must be a number");
  }
  // rest of the code
}

**Comment by taha-crafter** [comment]
Hi @wtitterton 

You've done an excellent job with this assignment, and you've met all the grading checklist 👍 

 just a few comments I left here.... and the fact that you could be using the Fake it strategy until you face 3 repetitions, then you make the obvious implementation.
 
 otherwise, everything else looks good to me, well done!

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts (hunk 2 above)
Comment (wtitterton) [reply 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts]: I think I had some issues passing a string.  It caused a typescript error when trying to test that scenario, so I just decided to check the input was within range. I think adding you above suggestion is a good idea though. This way we can be completely sure we are dealing with a number.

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts (hunk 1 above)
Comment (wtitterton) [reply 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts:5]: I agree, good spot. I had never heard of the each syntax in jest before this assignment. It's definitely been a game changer for me .

**Pull request merged**